modules/programs/wezterm.nix

```diff
@@ -0,0 +1,589 @@
+    mkKeyValue = key: value:
+      let
+        value' = if isString value then
+          ''"${value}"''
+        else if isBool value then
+          (boolToStr value)
+        else
+          toString value;
+      in "${key} = ${value'},";
+  };
+
+  toWeztermSettings = settings:
+    let
+      mapped = toWeztermSettings' settings;
+      indented = replaceStrings [ "\n" ] [ "\n  " ] mapped;
+      trimmed = removeSuffix "\n  " indented;
+    in ''
+      -- Settings
+        ${trimmed}'';
+
+  mkBindingMods = with types;
+    mkOption {
+      type = listOf
+        (enum [ "CTRL" "SUPER" "CMD" "WIN" "SHIFT" "ALT" "OPT" "LEADER" ]);
+      default = [ ];
+      description = ''
+        The binding modifier key(s).
+      '';
+      example = literalExample ''
+        [ "CTRL" ]
+      '';
+    };
+
+  mkBindingAction = with types;
+    example:
+    mkOption {
+      type = str;
+      default = "";
+      description = ''
+        The code called when the binding is executed.
+        You can set an action to <code>null</code> to disable the default assignment.
+      '';
+      example = example;
+    };
+
+  weztermKeybindType = with types;
+    submodule {
+      options = {
+        modifiers = mkBindingMods;
+
+        key = mkOption {
+          type = str;
+          description = ''
+            The keybinding main key.
+          '';
+          example = "l";
+        };
+
+        action = mkBindingAction
+          (literalExample "wezterm.action {ActivateTabRelative = 1}");
+      };
+    };
+
+  weztermMousebindType = with types;
+    submodule {
+      options = {
+        modifiers = mkBindingMods;
+
+        button = mkOption {
+          type = enum [ "Left" "Right" "Middle" ];
+          default = "Left";
+          description = ''
+            The mouse binding button.
+          '';
+          example = "Left";
+        };
+
+        event = mkOption {
+          type = enum [ "Up" "Down" "Drag" ];
+          default = "Left";
+          description = ''
+            The mouse binding event.
+          '';
+          example = "Left";
+        };
+
+        count = mkOption {
+          type = int;
+          default = 1;
+          description = ''
+            The event count to trigger action.
+          '';
+          example = 1;
+        };
+
+        action = mkBindingAction (literalExample ''"OpenLinkAtMouseCursor"'');
+      };
+    };
+
+  mkColor = default: desc:
+    mkOption {
+      type = types.str;
+      default = default;
+      description = desc;
+      example = default;
+    };
+
+  weztermColorschemeCursorType = with types;
+    submodule {
+      options = {
+        foreground = mkColor "#000000"
+          "Overrides the text color when the current cell is occupied by the cursor.";
+        background = mkColor "#52AD70"
+          "Overrides the cell background color when the current cell is occupied by the cursor and the cursor style is set to <literal>Block</literal>.";
```

Comment:
You can use a multiline string literal to avoid excessive line lengths. They start and end with `''`. Something like 
```suggestion
        background = mkColor "#52AD70" ''
          Overrides the cell background color when the current cell is occupied
          by the cursor and the cursor style is set to <literal>Block</literal>.
        '';
```
perhaps. Similar for the other long strings. Try to keep things at 80 characters wide or below.

Reply:
I may be wrong but they were originally multi-line strings and got converted after formatting. I've fixed them all now though, some of them are a little over 80 line but that's due to the `<literal>` tags.